### PR TITLE
Scripts: Make WSs get critical hit rate merits when applicable.

### PR DESCRIPTION
**Pull request by Hozu**
The current implementation neglected that.

### CODE REVIEW

File: scripts/globals/weaponskills.lua [hunk 1]
```diff
@@ -113,7 +113,7 @@ function doPhysicalWeaponskill(attacker, target, wsID, params, tp, primary)
             critrate = critrate + (10 + flourisheffect:getSubPower()/2)/100;
         end
         nativecrit = (attacker:getStat(MOD_DEX) - target:getStat(MOD_AGI))*0.005; -- assumes +0.5% crit rate per 1 dDEX
-        nativecrit = nativecrit + (attacker:getMod(MOD_CRITHITRATE)/100);
+        nativecrit = nativecrit + (attacker:getMod(MOD_CRITHITRATE)/100) + attacker:getMerit(MERIT_CRIT_HIT_RATE)/100;
```
Comment (bendangelo) [inline scripts/globals/weaponskills.lua]: I think you have to check it's a player because this will crash for mobs.

`attacker:getObjType() == TYPE_PC`

Comment (Hozu) [inline scripts/globals/weaponskills.lua]: You do in the core, but not lua. See the outdated comments here: https://github.com/DarkstarProject/darkstar/pull/2968

Comment (bendangelo) [inline scripts/globals/weaponskills.lua]: ok cool

**Comment by bendangelo** [comment]
Thanks for contributing :heart:

**Pull request merged**